src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -406,6 +414,135 @@ class CDCL {
         },
     )
 
+    // ---- Preprocessing ---- //
+
+    /**
+     * Preprocessing is ran before the main loop of the solver,
+     * allowing to spend some time simplifying the problem
+     * in exchange for a possibly faster solving time.
+     *
+     * @return if the solution is conclusive after preprocessing,
+     * return the model, otherwise return null
+     */
+    fun preprocess(): Model? {
```

Comment:
Consider NOT returning a Model. Either do not return anything, or consider returning `false` when the problem became (or initially was) UNSAT in the process of preprocessing, and `true` if the problem is SAT/UNKNOWN. Do not forget to document your decision.